flink-ml-lib/src/main/java/org/apache/flink/ml/feature/robustscaler/RobustScalerModel.java

```diff
@@ -0,0 +1,179 @@
+        final String broadcastModelKey = "broadcastModelKey";
+        DataStream<RobustScalerModelData> modelDataStream =
+                RobustScalerModelData.getModelDataStream(modelDataTable);
+
+        DataStream<Row> output =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(inputStream),
+                        Collections.singletonMap(broadcastModelKey, modelDataStream),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
+                            return inputData.map(
+                                    new PredictOutputFunction(
+                                            broadcastModelKey,
+                                            getInputCol(),
+                                            getWithCentering(),
+                                            getWithScaling()),
+                                    outputTypeInfo);
+                        });
+
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictOutputFunction extends RichMapFunction<Row, Row> {
+        private final String broadcastModelKey;
+        private final String inputCol;
+        private final boolean withCentering;
+        private final boolean withScaling;
+
+        private DenseVector medians;
+        private DenseVector scales;
+
+        public PredictOutputFunction(
+                String broadcastModelKey,
+                String inputCol,
+                boolean withCentering,
+                boolean withScaling) {
+            this.broadcastModelKey = broadcastModelKey;
+            this.inputCol = inputCol;
+            this.withCentering = withCentering;
+            this.withScaling = withScaling;
+        }
+
+        @Override
+        public Row map(Row row) throws Exception {
+            if (medians == null) {
+                RobustScalerModelData modelData =
+                        (RobustScalerModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastModelKey).get(0);
+                medians = modelData.medians;
+                scales =
+                        new DenseVector(
+                                Arrays.stream(modelData.ranges.values)
+                                        .map(range -> range == 0 ? 0 : 1 / range)
+                                        .toArray());
+            }
+            DenseVector outputVec = ((Vector) row.getField(inputCol)).clone().toDense();
+            Preconditions.checkState(
+                    medians.size() == outputVec.size(),
+                    "Number of features must be %s but got %s.",
+                    medians.size(),
+                    outputVec.size());
+
+            if (withCentering) {
+                BLAS.axpy(-1, medians, outputVec);
```

Comment:
Would it be better if we add the method `void axpy(double a, Vector x, Vector y)` and do `Vector`-to-`DenseVector` conversions in this method if it is the best practice?

Reply:
According to our offline discussion, I think that we should avoid allocating/deallocating memories in BLAS operations. Thus the current implementation is acceptable to me.